"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file

KINOVA (R) KORTEX (TM)

Copyright (c) 2018 Kinova inc. All rights reserved.

This software may be modified and distributed
under the terms of the BSD 3-Clause license.

Refer to the LICENSE file for details.
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import sys
import typing

if sys.version_info >= (3, 10):
    import typing as typing_extensions
else:
    import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor

class _ErrorCodes:
    ValueType = typing.NewType("ValueType", builtins.int)
    V: typing_extensions.TypeAlias = ValueType

class _ErrorCodesEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_ErrorCodes.ValueType], builtins.type):  # noqa: F821
    DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor
    ERROR_NONE: _ErrorCodes.ValueType  # 0
    """No error"""
    ERROR_PROTOCOL_SERVER: _ErrorCodes.ValueType  # 1
    """Protocol server error"""
    ERROR_PROTOCOL_CLIENT: _ErrorCodes.ValueType  # 2
    """Protocol client error"""
    ERROR_DEVICE: _ErrorCodes.ValueType  # 3
    """Device error"""
    ERROR_INTERNAL: _ErrorCodes.ValueType  # 4
    """Internal error"""

class ErrorCodes(_ErrorCodes, metaclass=_ErrorCodesEnumTypeWrapper):
    """Possible error codes"""

ERROR_NONE: ErrorCodes.ValueType  # 0
"""No error"""
ERROR_PROTOCOL_SERVER: ErrorCodes.ValueType  # 1
"""Protocol server error"""
ERROR_PROTOCOL_CLIENT: ErrorCodes.ValueType  # 2
"""Protocol client error"""
ERROR_DEVICE: ErrorCodes.ValueType  # 3
"""Device error"""
ERROR_INTERNAL: ErrorCodes.ValueType  # 4
"""Internal error"""
global___ErrorCodes = ErrorCodes

class _SubErrorCodes:
    ValueType = typing.NewType("ValueType", builtins.int)
    V: typing_extensions.TypeAlias = ValueType

class _SubErrorCodesEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_SubErrorCodes.ValueType], builtins.type):  # noqa: F821
    DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor
    SUB_ERROR_NONE: _SubErrorCodes.ValueType  # 0
    """No sub error"""
    METHOD_FAILED: _SubErrorCodes.ValueType  # 1
    """Method returned a failure status (generic error)"""
    UNIMPLEMENTED: _SubErrorCodes.ValueType  # 2
    """Unimplemented method"""
    INVALID_PARAM: _SubErrorCodes.ValueType  # 3
    """Invalid parameter"""
    UNSUPPORTED_SERVICE: _SubErrorCodes.ValueType  # 4
    """Service not recognized"""
    UNSUPPORTED_METHOD: _SubErrorCodes.ValueType  # 5
    """Method not recognized"""
    TOO_LARGE_ENCODED_FRAME_BUFFER: _SubErrorCodes.ValueType  # 6
    """Encoded frame bigger than what transport permits"""
    FRAME_ENCODING_ERR: _SubErrorCodes.ValueType  # 7
    """Unable to encode frame"""
    FRAME_DECODING_ERR: _SubErrorCodes.ValueType  # 8
    """Unable to decode frame"""
    INCOMPATIBLE_HEADER_VERSION: _SubErrorCodes.ValueType  # 9
    """Frame header version differs from what is expected and is considered incompatible"""
    UNSUPPORTED_FRAME_TYPE: _SubErrorCodes.ValueType  # 10
    """Unrecognized frame type"""
    UNREGISTERED_NOTIFICATION_RECEIVED: _SubErrorCodes.ValueType  # 11
    """Server receiving unregistered notification"""
    INVALID_SESSION: _SubErrorCodes.ValueType  # 12
    """Session not recognized"""
    PAYLOAD_DECODING_ERR: _SubErrorCodes.ValueType  # 13
    """Unable to decode payload"""
    UNREGISTERED_FRAME_RECEIVED: _SubErrorCodes.ValueType  # 14
    """Client received a response for which it did not send an RPC call"""
    INVALID_PASSWORD: _SubErrorCodes.ValueType  # 15
    """Password does not match specified user"""
    USER_NOT_FOUND: _SubErrorCodes.ValueType  # 16
    """Unrecognized user"""
    ENTITY_NOT_FOUND: _SubErrorCodes.ValueType  # 17
    """Cannot find entity"""
    ROBOT_MOVEMENT_IN_PROGRESS: _SubErrorCodes.ValueType  # 18
    """Robot refuses new control command because robot movement in progress"""
    ROBOT_NOT_MOVING: _SubErrorCodes.ValueType  # 19
    """Robot refuses stop command because robot is not moving"""
    NO_MORE_STORAGE_SPACE: _SubErrorCodes.ValueType  # 20
    """Unable to execute because no more storage"""
    ROBOT_NOT_READY: _SubErrorCodes.ValueType  # 21
    """Robot initialization is not complete"""
    ROBOT_IN_FAULT: _SubErrorCodes.ValueType  # 22
    """Robot in fault"""
    ROBOT_IN_MAINTENANCE: _SubErrorCodes.ValueType  # 23
    """Robot in maintenance"""
    ROBOT_IN_UPDATE_MODE: _SubErrorCodes.ValueType  # 24
    """Robot in update"""
    ROBOT_IN_EMERGENCY_STOP: _SubErrorCodes.ValueType  # 25
    """Robot in emergency stop state"""
    SINGLE_LEVEL_SERVOING: _SubErrorCodes.ValueType  # 26
    """Robot is in single-level servoing mode"""
    LOW_LEVEL_SERVOING: _SubErrorCodes.ValueType  # 27
    """Robot is in low-level servoing mode"""
    MAPPING_GROUP_NON_ROOT: _SubErrorCodes.ValueType  # 28
    """Trying to add a non-root MapGroup to Mapping"""
    MAPPING_INVALID_GROUP: _SubErrorCodes.ValueType  # 29
    """Trying to add an invalid or non-existent MapGroup to Mapping"""
    MAPPING_INVALID_MAP: _SubErrorCodes.ValueType  # 30
    """Trying to add an invalid or non-existent Map to Mapping"""
    MAP_GROUP_INVALID_MAP: _SubErrorCodes.ValueType  # 31
    """Trying to add an invalid or non-existent Map to MapGroup"""
    MAP_GROUP_INVALID_PARENT: _SubErrorCodes.ValueType  # 32
    """Trying to add a MapGroup under an invalid parent"""
    MAP_GROUP_INVALID_CHILD: _SubErrorCodes.ValueType  # 33
    """Trying to add an invalid or non-existent to MapGroup"""
    MAP_GROUP_INVALID_MOVE: _SubErrorCodes.ValueType  # 34
    """Trying to change a MapGroup's parent: move not supported"""
    MAP_IN_USE: _SubErrorCodes.ValueType  # 35
    """Deleting a Map used in a Mapping or MapGroup"""
    WIFI_CONNECT_ERROR: _SubErrorCodes.ValueType  # 36
    """Unable to connect to specified Wifi network"""
    UNSUPPORTED_NETWORK_TYPE: _SubErrorCodes.ValueType  # 37
    """Unsupported network type"""
    TOO_LARGE_ENCODED_PAYLOAD_BUFFER: _SubErrorCodes.ValueType  # 38
    """Encoded payload bigger than what transport permits"""
    UPDATE_PERMISSION_DENIED: _SubErrorCodes.ValueType  # 39
    """Attempting update command on non-updatable entity"""
    DELETE_PERMISSION_DENIED: _SubErrorCodes.ValueType  # 40
    """Attempting delete command on non-deletable entity"""
    DATABASE_ERROR: _SubErrorCodes.ValueType  # 41
    """Internal DB error"""
    UNSUPPORTED_OPTION: _SubErrorCodes.ValueType  # 42
    """Option not supported"""
    UNSUPPORTED_RESOLUTION: _SubErrorCodes.ValueType  # 43
    """Resolution not supported"""
    UNSUPPORTED_FRAME_RATE: _SubErrorCodes.ValueType  # 44
    """Frame rate not supported"""
    UNSUPPORTED_BIT_RATE: _SubErrorCodes.ValueType  # 45
    """Bit rate not supported"""
    UNSUPPORTED_ACTION: _SubErrorCodes.ValueType  # 46
    """Action not supported (generic, when an action is not supported for a particular item)"""
    UNSUPPORTED_FOCUS_ACTION: _SubErrorCodes.ValueType  # 47
    """Focus action not supported"""
    VALUE_IS_ABOVE_MAXIMUM: _SubErrorCodes.ValueType  # 48
    """Specified value is above the supported maximum"""
    VALUE_IS_BELOW_MINIMUM: _SubErrorCodes.ValueType  # 49
    """Specified value is below the supported minimum"""
    DEVICE_DISCONNECTED: _SubErrorCodes.ValueType  # 50
    """Device is not connected"""
    DEVICE_NOT_READY: _SubErrorCodes.ValueType  # 51
    """Device is not ready"""
    INVALID_DEVICE: _SubErrorCodes.ValueType  # 52
    """Device id is invalid during bridging"""
    SAFETY_THRESHOLD_REACHED: _SubErrorCodes.ValueType  # 53
    """Safety threshold is reached therefore safety is on"""
    INVALID_USER_SESSION_ACCESS: _SubErrorCodes.ValueType  # 54
    """Service or function access not allowed: out of session or level access"""
    CONTROL_MANUAL_STOP: _SubErrorCodes.ValueType  # 55
    """Manually stopped sequence or action"""
    CONTROL_OUTSIDE_WORKSPACE: _SubErrorCodes.ValueType  # 56
    """Commanded Cartesian position is outside of robot workspace"""
    CONTROL_ACTUATOR_COUNT_MISMATCH: _SubErrorCodes.ValueType  # 57
    """Number of constraint sent does not correspond to number of actuator (ex: joint speed)"""
    CONTROL_INVALID_DURATION: _SubErrorCodes.ValueType  # 58
    """Duration constraint is too short. The robot would need out of limit speeds/accelerations to reach this duration."""
    CONTROL_INVALID_SPEED: _SubErrorCodes.ValueType  # 59
    """Speed constraint is negative"""
    CONTROL_LARGE_SPEED: _SubErrorCodes.ValueType  # 60
    """Speed constraint is too high (exceed speed limit of leads to high acceleration)"""
    CONTROL_INVALID_ACCELERATION: _SubErrorCodes.ValueType  # 61
    """Speed constraint is too high or duration constraint too short and leads to high acceleration"""
    CONTROL_INVALID_TIME_STEP: _SubErrorCodes.ValueType  # 62
    """Refresh rate is smaller than the duration of the trajectory"""
    CONTROL_LARGE_SIZE: _SubErrorCodes.ValueType  # 63
    """Duration of the trajectory is more than 100s. The length of the trajectory is limited to 100000 points to avoid saturating the base memory."""
    CONTROL_WRONG_MODE: _SubErrorCodes.ValueType  # 64
    """Control mode is not a trajectory mode"""
    CONTROL_JOINT_POSITION_LIMIT: _SubErrorCodes.ValueType  # 65
    """Commanded configuration contains at least one actuator which is out of its physical limits"""
    CONTROL_NO_FILE_IN_MEMORY: _SubErrorCodes.ValueType  # 66
    """Trajectory is not computed and try to be started"""
    CONTROL_INDEX_OUT_OF_TRAJECTORY: _SubErrorCodes.ValueType  # 67
    """Attempting to read a point of the trajectory with an index higher than the number of point in trajectory point list."""
    CONTROL_ALREADY_RUNNING: _SubErrorCodes.ValueType  # 68
    """Trajectory is already running"""
    CONTROL_WRONG_STARTING_POINT: _SubErrorCodes.ValueType  # 69
    """Robot is not on the first point of the trajectory when we try to start the trajectory. This can happen if there is a motion between the moment when trajectory is computed and when it is started."""
    CONTROL_CARTESIAN_CANNOT_START: _SubErrorCodes.ValueType  # 70
    """Cannot start"""
    CONTROL_UNDEFINED_CONSTRAINT: _SubErrorCodes.ValueType  # 71
    """Kontrol library is not initialized"""
    CONTROL_UNINITIALIZED: _SubErrorCodes.ValueType  # 72
    """Contraint sent is not defined"""
    CONTROL_NO_ACTION: _SubErrorCodes.ValueType  # 73
    """Action does not exist"""
    CONTROL_UNDEFINED: _SubErrorCodes.ValueType  # 74
    """Undefined error"""
    WRONG_SERVOING_MODE: _SubErrorCodes.ValueType  # 75
    """Robot is in not in the right servoing mode"""
    CONTROL_WRONG_STARTING_SPEED: _SubErrorCodes.ValueType  # 76
    """Robot is not at the right speed when starting a new trajectory."""
    USERNAME_LENGTH_EXCEEDED: _SubErrorCodes.ValueType  # 100
    """User profile username length exceeds maximum allowed length"""
    FIRSTNAME_LENGTH_EXCEEDED: _SubErrorCodes.ValueType  # 101
    """User profile first name length exceeds maximum allowed length"""
    LASTNAME_LENGTH_EXCEEDED: _SubErrorCodes.ValueType  # 102
    """User profile last name length exceeds maximum allowed length"""
    PASSWORD_LENGTH_EXCEEDED: _SubErrorCodes.ValueType  # 103
    """User profile password length exceeds maximum allowed length"""
    USERNAME_ALREADY_EXISTS: _SubErrorCodes.ValueType  # 104
    """User profile username already in use by another profile"""
    USERNAME_EMPTY: _SubErrorCodes.ValueType  # 105
    """User profile empty username not allowed"""
    PASSWORD_NOT_CHANGED: _SubErrorCodes.ValueType  # 106
    """Change password both passwords are the same"""
    MAXIMUM_USER_PROFILES_USED: _SubErrorCodes.ValueType  # 107
    """Maximum number of user profiles in use"""
    ROUTER_UNVAILABLE: _SubErrorCodes.ValueType  # 108
    """The client router is currently unavailable. This can happen if an API method is called after the router has been deactivated via the method SetActivationStatus."""
    ADDRESS_NOT_IN_VALID_RANGE: _SubErrorCodes.ValueType  # 120
    """IP Address not valid against netmask"""
    ADDRESS_NOT_CONFIGURABLE: _SubErrorCodes.ValueType  # 121
    """IP Address not configurable on specified interface"""
    SESSION_NOT_IN_CONTROL: _SubErrorCodes.ValueType  # 130
    """Trying to perform command from a non-controlling session in single-level mode"""
    METHOD_TIMEOUT: _SubErrorCodes.ValueType  # 131
    """Timeout occured during method execution"""
    UNSUPPORTED_ROBOT_CONFIGURATION: _SubErrorCodes.ValueType  # 132
    """Product Configuration setter method failed because changing this parameter is unsupported on your robot model"""
    NVRAM_READ_FAIL: _SubErrorCodes.ValueType  # 133
    """Failed to read in NVRAM."""
    NVRAM_WRITE_FAIL: _SubErrorCodes.ValueType  # 134
    """Failed to write in NVRAM."""
    NETWORK_NO_ADDRESS_ASSIGNED: _SubErrorCodes.ValueType  # 135
    """The specified interface has no assigned IP"""
    READ_PERMISSION_DENIED: _SubErrorCodes.ValueType  # 136
    """Attempting read command on unreadable entity"""
    CONTROLLER_INVALID_MAPPING: _SubErrorCodes.ValueType  # 137
    """Attempting to assign an unsuited mapping to controller"""
    ACTION_IN_USE: _SubErrorCodes.ValueType  # 138
    """Attempting to delete an Action used by another entity"""
    SEND_FAILED: _SubErrorCodes.ValueType  # 139
    """An error occured when trying to send the message (Transport error)"""
    CONTROL_WAYPOINT_TRAJECTORY_ABORTED: _SubErrorCodes.ValueType  # 140
    """Waypoint trajectory sequence aborted by kontrol library"""
    CONTROL_PERMISSION_DENIED: _SubErrorCodes.ValueType  # 141
    """Permission to execute a routine from the kontrol interface denied"""

class SubErrorCodes(_SubErrorCodes, metaclass=_SubErrorCodesEnumTypeWrapper):
    """Possible sub-error codes"""

SUB_ERROR_NONE: SubErrorCodes.ValueType  # 0
"""No sub error"""
METHOD_FAILED: SubErrorCodes.ValueType  # 1
"""Method returned a failure status (generic error)"""
UNIMPLEMENTED: SubErrorCodes.ValueType  # 2
"""Unimplemented method"""
INVALID_PARAM: SubErrorCodes.ValueType  # 3
"""Invalid parameter"""
UNSUPPORTED_SERVICE: SubErrorCodes.ValueType  # 4
"""Service not recognized"""
UNSUPPORTED_METHOD: SubErrorCodes.ValueType  # 5
"""Method not recognized"""
TOO_LARGE_ENCODED_FRAME_BUFFER: SubErrorCodes.ValueType  # 6
"""Encoded frame bigger than what transport permits"""
FRAME_ENCODING_ERR: SubErrorCodes.ValueType  # 7
"""Unable to encode frame"""
FRAME_DECODING_ERR: SubErrorCodes.ValueType  # 8
"""Unable to decode frame"""
INCOMPATIBLE_HEADER_VERSION: SubErrorCodes.ValueType  # 9
"""Frame header version differs from what is expected and is considered incompatible"""
UNSUPPORTED_FRAME_TYPE: SubErrorCodes.ValueType  # 10
"""Unrecognized frame type"""
UNREGISTERED_NOTIFICATION_RECEIVED: SubErrorCodes.ValueType  # 11
"""Server receiving unregistered notification"""
INVALID_SESSION: SubErrorCodes.ValueType  # 12
"""Session not recognized"""
PAYLOAD_DECODING_ERR: SubErrorCodes.ValueType  # 13
"""Unable to decode payload"""
UNREGISTERED_FRAME_RECEIVED: SubErrorCodes.ValueType  # 14
"""Client received a response for which it did not send an RPC call"""
INVALID_PASSWORD: SubErrorCodes.ValueType  # 15
"""Password does not match specified user"""
USER_NOT_FOUND: SubErrorCodes.ValueType  # 16
"""Unrecognized user"""
ENTITY_NOT_FOUND: SubErrorCodes.ValueType  # 17
"""Cannot find entity"""
ROBOT_MOVEMENT_IN_PROGRESS: SubErrorCodes.ValueType  # 18
"""Robot refuses new control command because robot movement in progress"""
ROBOT_NOT_MOVING: SubErrorCodes.ValueType  # 19
"""Robot refuses stop command because robot is not moving"""
NO_MORE_STORAGE_SPACE: SubErrorCodes.ValueType  # 20
"""Unable to execute because no more storage"""
ROBOT_NOT_READY: SubErrorCodes.ValueType  # 21
"""Robot initialization is not complete"""
ROBOT_IN_FAULT: SubErrorCodes.ValueType  # 22
"""Robot in fault"""
ROBOT_IN_MAINTENANCE: SubErrorCodes.ValueType  # 23
"""Robot in maintenance"""
ROBOT_IN_UPDATE_MODE: SubErrorCodes.ValueType  # 24
"""Robot in update"""
ROBOT_IN_EMERGENCY_STOP: SubErrorCodes.ValueType  # 25
"""Robot in emergency stop state"""
SINGLE_LEVEL_SERVOING: SubErrorCodes.ValueType  # 26
"""Robot is in single-level servoing mode"""
LOW_LEVEL_SERVOING: SubErrorCodes.ValueType  # 27
"""Robot is in low-level servoing mode"""
MAPPING_GROUP_NON_ROOT: SubErrorCodes.ValueType  # 28
"""Trying to add a non-root MapGroup to Mapping"""
MAPPING_INVALID_GROUP: SubErrorCodes.ValueType  # 29
"""Trying to add an invalid or non-existent MapGroup to Mapping"""
MAPPING_INVALID_MAP: SubErrorCodes.ValueType  # 30
"""Trying to add an invalid or non-existent Map to Mapping"""
MAP_GROUP_INVALID_MAP: SubErrorCodes.ValueType  # 31
"""Trying to add an invalid or non-existent Map to MapGroup"""
MAP_GROUP_INVALID_PARENT: SubErrorCodes.ValueType  # 32
"""Trying to add a MapGroup under an invalid parent"""
MAP_GROUP_INVALID_CHILD: SubErrorCodes.ValueType  # 33
"""Trying to add an invalid or non-existent to MapGroup"""
MAP_GROUP_INVALID_MOVE: SubErrorCodes.ValueType  # 34
"""Trying to change a MapGroup's parent: move not supported"""
MAP_IN_USE: SubErrorCodes.ValueType  # 35
"""Deleting a Map used in a Mapping or MapGroup"""
WIFI_CONNECT_ERROR: SubErrorCodes.ValueType  # 36
"""Unable to connect to specified Wifi network"""
UNSUPPORTED_NETWORK_TYPE: SubErrorCodes.ValueType  # 37
"""Unsupported network type"""
TOO_LARGE_ENCODED_PAYLOAD_BUFFER: SubErrorCodes.ValueType  # 38
"""Encoded payload bigger than what transport permits"""
UPDATE_PERMISSION_DENIED: SubErrorCodes.ValueType  # 39
"""Attempting update command on non-updatable entity"""
DELETE_PERMISSION_DENIED: SubErrorCodes.ValueType  # 40
"""Attempting delete command on non-deletable entity"""
DATABASE_ERROR: SubErrorCodes.ValueType  # 41
"""Internal DB error"""
UNSUPPORTED_OPTION: SubErrorCodes.ValueType  # 42
"""Option not supported"""
UNSUPPORTED_RESOLUTION: SubErrorCodes.ValueType  # 43
"""Resolution not supported"""
UNSUPPORTED_FRAME_RATE: SubErrorCodes.ValueType  # 44
"""Frame rate not supported"""
UNSUPPORTED_BIT_RATE: SubErrorCodes.ValueType  # 45
"""Bit rate not supported"""
UNSUPPORTED_ACTION: SubErrorCodes.ValueType  # 46
"""Action not supported (generic, when an action is not supported for a particular item)"""
UNSUPPORTED_FOCUS_ACTION: SubErrorCodes.ValueType  # 47
"""Focus action not supported"""
VALUE_IS_ABOVE_MAXIMUM: SubErrorCodes.ValueType  # 48
"""Specified value is above the supported maximum"""
VALUE_IS_BELOW_MINIMUM: SubErrorCodes.ValueType  # 49
"""Specified value is below the supported minimum"""
DEVICE_DISCONNECTED: SubErrorCodes.ValueType  # 50
"""Device is not connected"""
DEVICE_NOT_READY: SubErrorCodes.ValueType  # 51
"""Device is not ready"""
INVALID_DEVICE: SubErrorCodes.ValueType  # 52
"""Device id is invalid during bridging"""
SAFETY_THRESHOLD_REACHED: SubErrorCodes.ValueType  # 53
"""Safety threshold is reached therefore safety is on"""
INVALID_USER_SESSION_ACCESS: SubErrorCodes.ValueType  # 54
"""Service or function access not allowed: out of session or level access"""
CONTROL_MANUAL_STOP: SubErrorCodes.ValueType  # 55
"""Manually stopped sequence or action"""
CONTROL_OUTSIDE_WORKSPACE: SubErrorCodes.ValueType  # 56
"""Commanded Cartesian position is outside of robot workspace"""
CONTROL_ACTUATOR_COUNT_MISMATCH: SubErrorCodes.ValueType  # 57
"""Number of constraint sent does not correspond to number of actuator (ex: joint speed)"""
CONTROL_INVALID_DURATION: SubErrorCodes.ValueType  # 58
"""Duration constraint is too short. The robot would need out of limit speeds/accelerations to reach this duration."""
CONTROL_INVALID_SPEED: SubErrorCodes.ValueType  # 59
"""Speed constraint is negative"""
CONTROL_LARGE_SPEED: SubErrorCodes.ValueType  # 60
"""Speed constraint is too high (exceed speed limit of leads to high acceleration)"""
CONTROL_INVALID_ACCELERATION: SubErrorCodes.ValueType  # 61
"""Speed constraint is too high or duration constraint too short and leads to high acceleration"""
CONTROL_INVALID_TIME_STEP: SubErrorCodes.ValueType  # 62
"""Refresh rate is smaller than the duration of the trajectory"""
CONTROL_LARGE_SIZE: SubErrorCodes.ValueType  # 63
"""Duration of the trajectory is more than 100s. The length of the trajectory is limited to 100000 points to avoid saturating the base memory."""
CONTROL_WRONG_MODE: SubErrorCodes.ValueType  # 64
"""Control mode is not a trajectory mode"""
CONTROL_JOINT_POSITION_LIMIT: SubErrorCodes.ValueType  # 65
"""Commanded configuration contains at least one actuator which is out of its physical limits"""
CONTROL_NO_FILE_IN_MEMORY: SubErrorCodes.ValueType  # 66
"""Trajectory is not computed and try to be started"""
CONTROL_INDEX_OUT_OF_TRAJECTORY: SubErrorCodes.ValueType  # 67
"""Attempting to read a point of the trajectory with an index higher than the number of point in trajectory point list."""
CONTROL_ALREADY_RUNNING: SubErrorCodes.ValueType  # 68
"""Trajectory is already running"""
CONTROL_WRONG_STARTING_POINT: SubErrorCodes.ValueType  # 69
"""Robot is not on the first point of the trajectory when we try to start the trajectory. This can happen if there is a motion between the moment when trajectory is computed and when it is started."""
CONTROL_CARTESIAN_CANNOT_START: SubErrorCodes.ValueType  # 70
"""Cannot start"""
CONTROL_UNDEFINED_CONSTRAINT: SubErrorCodes.ValueType  # 71
"""Kontrol library is not initialized"""
CONTROL_UNINITIALIZED: SubErrorCodes.ValueType  # 72
"""Contraint sent is not defined"""
CONTROL_NO_ACTION: SubErrorCodes.ValueType  # 73
"""Action does not exist"""
CONTROL_UNDEFINED: SubErrorCodes.ValueType  # 74
"""Undefined error"""
WRONG_SERVOING_MODE: SubErrorCodes.ValueType  # 75
"""Robot is in not in the right servoing mode"""
CONTROL_WRONG_STARTING_SPEED: SubErrorCodes.ValueType  # 76
"""Robot is not at the right speed when starting a new trajectory."""
USERNAME_LENGTH_EXCEEDED: SubErrorCodes.ValueType  # 100
"""User profile username length exceeds maximum allowed length"""
FIRSTNAME_LENGTH_EXCEEDED: SubErrorCodes.ValueType  # 101
"""User profile first name length exceeds maximum allowed length"""
LASTNAME_LENGTH_EXCEEDED: SubErrorCodes.ValueType  # 102
"""User profile last name length exceeds maximum allowed length"""
PASSWORD_LENGTH_EXCEEDED: SubErrorCodes.ValueType  # 103
"""User profile password length exceeds maximum allowed length"""
USERNAME_ALREADY_EXISTS: SubErrorCodes.ValueType  # 104
"""User profile username already in use by another profile"""
USERNAME_EMPTY: SubErrorCodes.ValueType  # 105
"""User profile empty username not allowed"""
PASSWORD_NOT_CHANGED: SubErrorCodes.ValueType  # 106
"""Change password both passwords are the same"""
MAXIMUM_USER_PROFILES_USED: SubErrorCodes.ValueType  # 107
"""Maximum number of user profiles in use"""
ROUTER_UNVAILABLE: SubErrorCodes.ValueType  # 108
"""The client router is currently unavailable. This can happen if an API method is called after the router has been deactivated via the method SetActivationStatus."""
ADDRESS_NOT_IN_VALID_RANGE: SubErrorCodes.ValueType  # 120
"""IP Address not valid against netmask"""
ADDRESS_NOT_CONFIGURABLE: SubErrorCodes.ValueType  # 121
"""IP Address not configurable on specified interface"""
SESSION_NOT_IN_CONTROL: SubErrorCodes.ValueType  # 130
"""Trying to perform command from a non-controlling session in single-level mode"""
METHOD_TIMEOUT: SubErrorCodes.ValueType  # 131
"""Timeout occured during method execution"""
UNSUPPORTED_ROBOT_CONFIGURATION: SubErrorCodes.ValueType  # 132
"""Product Configuration setter method failed because changing this parameter is unsupported on your robot model"""
NVRAM_READ_FAIL: SubErrorCodes.ValueType  # 133
"""Failed to read in NVRAM."""
NVRAM_WRITE_FAIL: SubErrorCodes.ValueType  # 134
"""Failed to write in NVRAM."""
NETWORK_NO_ADDRESS_ASSIGNED: SubErrorCodes.ValueType  # 135
"""The specified interface has no assigned IP"""
READ_PERMISSION_DENIED: SubErrorCodes.ValueType  # 136
"""Attempting read command on unreadable entity"""
CONTROLLER_INVALID_MAPPING: SubErrorCodes.ValueType  # 137
"""Attempting to assign an unsuited mapping to controller"""
ACTION_IN_USE: SubErrorCodes.ValueType  # 138
"""Attempting to delete an Action used by another entity"""
SEND_FAILED: SubErrorCodes.ValueType  # 139
"""An error occured when trying to send the message (Transport error)"""
CONTROL_WAYPOINT_TRAJECTORY_ABORTED: SubErrorCodes.ValueType  # 140
"""Waypoint trajectory sequence aborted by kontrol library"""
CONTROL_PERMISSION_DENIED: SubErrorCodes.ValueType  # 141
"""Permission to execute a routine from the kontrol interface denied"""
global___SubErrorCodes = SubErrorCodes
